gateway/ln-gateway/src/client.rs

```diff
@@ -47,30 +46,47 @@ impl StandardGatewayClientBuilder {
         lnrpc: Arc<dyn ILnRpcClient>,
         old_client: Option<fedimint_client::Client>,
     ) -> Result<fedimint_client::Client> {
-        let federation_id = config.config.global.federation_id;
+        let FederationConfig {
+            invite_code,
+            mint_channel_id,
+            timelock_delta,
+            fees,
+        } = config;
+        let federation_id = invite_code.id;
 
         let mut registry = self.registry.clone();
         registry.attach(GatewayClientGen {
             lnrpc,
             node_pub_key,
             lightning_alias,
-            fees: config.fees,
-            timelock_delta: config.timelock_delta,
-            mint_channel_id: config.mint_channel_id,
+            fees,
+            timelock_delta,
+            mint_channel_id,
         });
 
         let mut client_builder = ClientBuilder::default();
         client_builder.with_module_inits(registry);
         client_builder.with_primary_module(self.primary_module);
-        client_builder.with_config(config.config);
         if let Some(old_client) = old_client {
             client_builder.with_old_client_database(old_client);
         } else {
             let db_path = self.work_dir.join(format!("{federation_id}.db"));
-            let db = fedimint_rocksdb::RocksDb::open(db_path).map_err(|e| {
+            {
+                let rocksdb = fedimint_rocksdb::RocksDb::open(db_path.clone()).map_err(|e| {
+                    GatewayError::DatabaseError(anyhow::anyhow!("Error opening rocksdb: {e:?}"))
+                })?;
+
+                // Initialize a client database to check if a config was previously saved in it
+                let db = Database::new(rocksdb, ModuleDecoderRegistry::default());
+                if (get_config_from_db(&db).await).is_none() {
+                    client_builder.with_invite_code(invite_code);
+                }
+            }
+
+            let rocksdb = fedimint_rocksdb::RocksDb::open(db_path.clone()).map_err(|e| {
```

Comment:
Nit: Do we need to open a second handle to `rocksdb` or is it possible to just open it once, check if the config exists, then add the db using `with_database`?

Reply:
I couldn't figure how to keep that first rocksdb handle after consuming it and checking if config is in db
```rs
    let db = Database::new(rocksdb, ModuleDecoderRegistry::default());
    if (get_config_from_db(&db).await).is_none() {
        client_builder.with_invite_code(invite_code);
    }
```

Reply:
Any suggestions on how to do this?

Reply:
I don't think rocksdb is cloneable so this might not be easy to do, I think its fine as is then